byte_tester/main.c

```diff
@@ -140,15 +140,89 @@ int _read_binary_file(char *filename, int *did_we_read_data, FILE *f, int *file_
   return SUCCEEDED;
 }
 
+/* Get the next token from the input string and write the results on
+   the result argument.  */
+int get_token(char **input, char *result) {
+  int n = 0;
+
+  sscanf(*input, "%255s%n", result, &n);
+
+  if (!n) {
+    return FAILED;
+  }
+
+  *input += n;
+  return SUCCEEDED;
+}
+
+/* Extracts contents of byte tester comments in-place. The result is
+   placed on the input string itself, overwriting its contents.  */
+int extract_comments(char *input, size_t size) {
+  char *current = input, *output = input, *end = NULL;
+
+  while (*current) {
+    int comment_length;
+    char *comment = NULL, *result = NULL;
+
+    /* Search for the next comment of
+       each type, picking the nearest.  */
+
+    /* Semicolon comment */
+    result = strstr(current, "; @BT");
+    if (result) {
+      comment = result + 5;
+      end = strchr(comment, '\n');
+    }
+
+    /* Double slash comment */
+    result = strstr(current, "// @BT");
+    if (result && (!comment || result < comment)) {
+      comment = result + 6;
+      end = strchr(comment, '\n');
+    }
+
+    /* Block comment */
+    result = strstr(current, "/* @BT");
+    if (result && (!comment || result < comment)) {
+      comment = result + 6;
+      end = strstr(comment, "*/");
+    }
+
+    /* Stop parsing if there is no more comments */
+    if (!comment) {
+      break;
+    }
+
+    /* Extend unclosed comments to the end of the input string.
+       This handles the cases where a semicolon or double slash
+       comment is on the last line, or a unclosed block comment. */
+    if (!end) {
+      end = input + size;
+    }
+
+    /* Overwrite input string with comment contents */
+    current = comment;
+    comment_length = end - current;
+
+    memmove(output, current, comment_length);
+    output += comment_length;
+    current += comment_length;
+  }
+
+  *output = '\0';
+
+  return SUCCEEDED;
+}
 
 int main(int argc, char *argv[]) {
 
   char tmp[256], test_id[256], tag_id[256];
+  char *input_name = NULL, *input = NULL, *current = NULL;
   unsigned char bytes[256];
-  int file_size, end, byte_count, i, j, length, tag_start, tag_end, wrong_bytes, failures, use_address = NO, address = 0, did_we_read_data = NO, got_it = NO;
+  int input_size, file_size, end, byte_count, i, j, length, tag_start, tag_end, wrong_bytes, failures, should_extract_comments = NO, use_address = NO, address = 0, did_we_read_data = NO, got_it = NO;
   FILE *f;
   
-  if (argc != 2 || argv == NULL) {
+  if (argc < 2 || argv == NULL) {
```

Comment:
Why we allow e.g., 10 arguments while only "-s" and filename are used?

Reply:
Hmm, you are right.

We can allow only two or three arguments:
```c
if ((argc != 2 && argc != 3) || argv == NULL) {
```

Or a range where  `1 < argc < 4`, that doesn't makes much sense right now but could be useful in the future when more args gets added:
```c
if (argc < 2 || argc > 3 || argv == NULL) {
```

Which one do you prefer?

Reply:
Both are fine, but perhaps 

if (argc < 2 || argc > 3 || argv == NULL) {

is minimally better? :)

Reply:
I agree :)

Done.